src/Features/Core/Portable/PdbSourceDocument/PdbSourceDocumentMetadataAsSourceFileProvider.cs

```diff
@@ -237,7 +248,8 @@ public bool TryRemoveDocumentFromWorkspace(Workspace workspace, string filePath)
         {
             if (_fileToDocumentMap.TryGetValue(filePath, out var documentId))
             {
-                workspace.OnDocumentClosed(documentId, new FileTextLoader(filePath, MetadataAsSourceGeneratedFileInfo.Encoding));
+                var document = workspace.CurrentSolution.GetDocument(documentId);
+                workspace.OnDocumentClosed(documentId, new FileTextLoader(filePath, document?.GetTextSynchronously(CancellationToken.None).Encoding ?? Encoding.UTF8));
```

Comment:
>GetTextSynchronously [](http://example.com/codeflow?start=94&length=20)

why synchronous?

Reply:
This is called on the UI thread from an RDT event handler.

Reply:
Let's add a comment. Would also be nice if we didn't need to fetch the text just so we can get to the encoding. @jasonmalinowski

Reply:
How would we know the encoding without getting the text?

Reply:
(I haven't reviewed the larger PR yet, so if there's context I'm missing tell me to just review the full thing)

Reply:
The existing text loader does not know the encoding?

Reply:
There's no remembering of the encoding from a previous operation, no. They are just used to populate the existing text object, and then anything is dropped from them.